### DISCUSSION TITLE
Honeycomb Bravo Throttle

### DISCUSSION THREAD

**Discussion by Rosmanczyk** [flybywiresim/aircraft#5210] [Help]
Is there a configuration guide for making Honeycombs Bravo Throttle work with the A32NX? Or is it not possible at all at the moment? I can't seem to make the autopilot modes work, especially following the flight plan in NAV mode.

**Comment by tracernz** [comment]
For dev version:
You have to decide where you want the detents (since there are no hardware ones on the Bravo); I suggest the back of the lever aligned with each of the marks next to the levers is a good place. Disable reversers on axis (since Bravo doesn't have reverse axis), and calibrate each detent on our EFB, where you decided they should be. For reverse you'll want to bind the reverse "button" to the decrease throttle button in MSFS settings, as for any other aircraft.

**Reply by Rosmanczyk** [reply]
Hey there, thank you a lot for your help! I have the detents and throttles set up and they work without a problem. I just can't make the autopilot functions work. This is something I'd like to know if there is a way to use Honeycomb Bravo Throttle's autopilot buttons and mapping them to work with the A32NX.

**Reply by tracernz** [reply]
Ah, got you. There are a bunch of custom events to drive the FCU and autopilot, documented in https://github.com/flybywiresim/a32nx/blob/master/docs/a320-events.md. Does Honeycomb have a software that can map these? I know they do for X-Plane.

**Reply by Rosmanczyk** [reply]
Thanks for your help! I don't know if there's something like it for msfs 2020. Even if it was I wouldn't know how to map things via scripts.

**Reply by tracernz** [reply]
> 
> 
> Thanks for your help! I don't know if there's something like it for msfs 2020. Even if it was I wouldn't know how to map things via scripts.

I believe it can be done with FSUIPC or SPAD.Next, but I don't have any personal experience with either and they're both payware. We have some building blocks for a free and open source solution but no firm plans to actually implement the rest yet, so don't count on that.

**Reply by aguther** [reply]
I have the Honeycomb Bravo myself and I was able to get it working both with SPAD.neXt and FSUIPC, maybe this is helping you:
https://docs.flybywiresim.com/start/autopilot-fbw/#mapping-of-events-to-control-autopilot-autothrust

**Reply by cadilhac** [reply]
Unfortunately, using the profile for spadnext and the corresponding new events, or sending regular PLUS/MINUS events, I always have an issue: when turning the knob quickly it misses most of the clicks and it makes rotating the knob very tedious/long.

**Reply by Rosmanczyk** [reply]
I don't use the PLUS/MINUS events in MSFS. For the FBW I have SpadNext setup so that it increments the altitude hold value by either 100 or by 1000 depending on if the ALT HOLD value surpasses a certain range. (in lower ranges increment and decrement by 100 and in higher ranges by 1000 respectively). It's also possible to do the same via the interval knob that changes 100s and 1000s or vice versa. But I didn't want to fiddle around with another knob that I'd have to control.

**Reply by cadilhac** [reply]
> I don't use the PLUS/MINUS events in MSFS. For the FBW I have SpadNext setup so that it increments the altitude hold value by either 100 or by 1000 depending on if the ALT HOLD value surpasses a certain range. (in lower ranges increment and decrement by 100 and in higher ranges by 1000 respectively). It's also possible to do the same via the interval knob that changes 100s and 1000s or vice versa. But I didn't want to fiddle around with another knob that I'd have to control.

hummm, ok, maybe, but it doesn't solve the speed/acceleration issue of the knob.
It's as if MSFS was not able to detect/process the events quickly enough to be able to handle them all.

**Reply by Rosmanczyk** [reply]
It is also possible solving this problem in SpadNext! I've seen it in some profiles where there is a retriggerable delay that resets when you stop turning the knob. When you turn it in a certain speed SpadNext (within a certain time range) let's you configure what should happen when you turn it fast. 
Check out Les O'Reilly on YouTube, he uses this technique in one of his videos, I think it's on the TBM or so at 3:10:

https://www.youtube.com/watch?v=7U9YPcLAg7Y&t=213s

**Reply by cadilhac** [reply]
I imagined that there might be a solution like that. Thanks for the link, I'll go check this.

**Reply by cadilhac** [reply]
Unfortunately, O'Reilly's technique doesn't work well with events (he says so in the comments but for me it works but with the delay I explains below). So I searched a variable, but I only found A32NX_AUTOPILOT_HEADING_SELECTED. It correctly changes the heading on the ND, but not on the FCU although the [doc](https://docs.flybywiresim.com/fbw-a32nx/feature-guides/autopilot-fbw/#speed) says this variable is "Selected Heading in FCU"...

Also the delay he defines is a true pause (spadnext does nothing during this delay) and not a real timer. This produces a lag (one click produces the increment only after the 200ms delay). Programmatically speaking, there should be a timer that fires 200ms later and if during this delay a click is done again, the timer would be reset and started again. But this feature does not exist in spadnext I guess...

**Reply by Rosmanczyk** [reply]
I think this is something Honeycomb would need to implement into their encoder or so.
For the ALTITUDE selection to work you need two variables set up:
1.  A32NX.FCU_ALT_INC or A32NX.FCU_ALT_DEC and
2. HEADING_BUG_INC or HEADING_BUG_DEC

When both of these are triggered then both will update accordingly in the ND and FCU.

**Reply by cadilhac** [reply]
Not sure what you mean. These are events, not variables. And you mix heading and alt + default events and custom events.

**Comment by Rosmanczyk** [comment]
Oh yes sorry my bad I meant A32NX.FCU_HDG_INC or DEC and HEADING_BUG_INC or DEC.
These are two events that are fired on my end when I want to manipulate the heading and in both places it updates correctly.

**Reply by cadilhac** [reply]
Do you mean it updates correctly with O'Reilly's technique, by increment of 1 or 10 and instantly?

**Reply by Rosmanczyk** [reply]
Yes it works just fine for me. Dialing in a 360 degree turn is super quickly accomplished and normally when manipulating the heading I don't freak out on the speed of the dial anyway. Been flying with this setup and it's super easy. (to me at least)

**Reply by cadilhac** [reply]
Sorry, I don't get it. If I fire both A32NX.FCU_HDG_INC and HEADING_BUG_INC, one notch will simply increase the heading by 2°.

**Reply by Rosmanczyk** [reply]
Did you delete the PLUS/MINUS binding in MSFS itself? Easily forgotten sometimes

**Reply by cadilhac** [reply]
Unfortunately, yes, I already removed them. I would have liked it to be as simple as this.

**Reply by Rosmanczyk** [reply]
what hardware are you using and is there a way you could show me your spad.next setup for the buttons? hard to tell otherwise what is happening

**Reply by Rosmanczyk** [reply]
if you want we can continue chatting somewhere else if that would help you

**Reply by cadilhac** [reply]
For me these events are the same and therefore do twice the job.
Also there is no way to increment by 10 with such events.
Thanks for the chat proposition. But where/when?

![image](https://user-images.githubusercontent.com/50608/149237820-e487c746-5d88-4406-b458-2bb09c91e7e0.png)

**Reply by Rosmanczyk** [reply]
Just choose anything you might want to use for further communication I'm pretty open minded there.

**Reply by Rosmanczyk** [reply]
Actually I just checked my SpadNext profile here and I don't use any A32NX.FCU event for the heading..I just use "HEADING_BUG_INC" oder DEC

**Reply by cadilhac** [reply]
Ah, this makes sense now. But this is an event to increase by 1. Setting a param of 10 does not work.

**Reply by Rosmanczyk** [reply]
I think you're not gonna like this but just create the event 10 times when triggered. All of this fires so quickly it's not gonna bother your computer anyway. It's not the most elegant of solutions but it works and it still gives you more freedom than MSFS does out of the box.

**Reply by cadilhac** [reply]
I like everything that is working in the end. ;)
So, it kinda works. But adding 10 events makes the knob increase like crazy! Surprisingly, it would need less to get a good speed pattern.
Thanks anyway for all this.

And if one of the devs wants to explain why the A32NX_AUTOPILOT_HEADING_SELECTED variable does update only the ND and not the FCU, feel free to shed light on this.

**Reply by Rosmanczyk** [reply]
I guess one of those variables is the custom autopilot they created and the other a build in MSFS function? just guessing here.
Yeah ten times the event is much, I used 3 and that's more than enough, maybe even two could suffice.
You're welcome I hope I could help a bit.

**Reply by tracernz** [reply]
> I like everything that is working in the end. ;) So, it kinda works. But adding 10 events makes the knob increase like crazy! Surprisingly, it would need less to get a good speed pattern. Thanks anyway for all this.
> 
> And if one of the devs wants to explain why the A32NX_AUTOPILOT_HEADING_SELECTED variable does update only the ND and not the FCU, feel free to shed light on this.

Because the variable is intended to be read-only (only written by the FCU). You should use `K:A32NX.FCU_HDG_INC`, `K:A32NX.FCU_HDG_DEC`, or `K:A32NX.FCU_HDG_SET` with the desired heading as the param in the latter case.
